pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go

```diff
@@ -126,13 +128,12 @@ func (c *StaticPodStateController) sync(ctx context.Context, syncCtx factory.Syn
 	}
 
 	switch {
-	case len(images) == 0:
-		// if NodeStatuses is still empty, we are most probably in bootstrapping phase and this controller races with the
-		// installer controller. Hence, ignore that case. It will settle.
-
-		if len(originalOperatorStatus.NodeStatuses) > 0 {
-			syncCtx.Recorder().Warningf("MissingVersion", "no image found for operand pod")
-		}
+	// if NodeStatuses is still empty, we are most probably in bootstrapping phase and this controller races with the
+	// installer controller.
+	// If we can't Get any Pod from the apiserver most probably there is a network problem.
+	// Hence, ignore that case. It will settle.
```

Comment:
the argument is not about network problem, but about having found at least one pod, but it lacks the env var for the image. This network comment is more confusing. Would just say that "// we found pods, but none has an image".

Reply:
yep, actually I have to handle all the len(images) == 0 cases here otherwise it hits the default clause.